pennylane/templates/subroutines/adder.py

```diff
@@ -0,0 +1,114 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the Adder template.
+"""
+
+import pennylane as qml
+from pennylane.operation import Operation
+
+
+class Adder(Operation):
+    r"""Performs the Inplace Addition operation.
+
+    This operator adds the integer :math:`k` modulo :math:`mod` in the computational basis:
+
+    .. math::
+
+        \text{Adder}(k,mod) |m \rangle = | m+k \, \text{mod} \, mod \rangle,
+
+    The decomposition of this operator is based on the QFT-based method presented in `Atchade-Adelomou and Gonzalez (2023) <https://arxiv.org/abs/2311.08555>`_.
+
+        Args:
+            k (int): number that wants to be added.
+            wires (Sequence[int]): the wires the operation acts on. There are needed at least enough wires to represent k and mod.
+            mod (int): modulo with respect to which the sum is performed, default value will be ``2^len(wires)``.
+            work_wires (Sequence[int]): the auxiliary wires to use for the sum modulo :math:`mod` when :math:`mod \neq 2^{\text{len(wires)}}`.
+
+        **Example**
+
+        Sum of two integers :math:`m=8` and :math:`k=5` modulo :math:`mod=15`. Note that to perform this sum using qml.Adder we need that :math:`m,k < mod`.
+
+        .. code-block::
+
+            m = 8
+            k = 5
+            mod = 15
+            wires_m =[0,1,2,3]
+            work_wires=[4,5]
+            dev = qml.device("default.qubit", shots=1)
+            @qml.qnode(dev)
+            def adder_modulo(m, k, mod, wires_m, work_wires):
+                # Function that performs m + k modulo mod in the computational basis
+                qml.BasisEmbedding(m, wires=wires_m)
+                qml.Adder(k, wires_m, mod, work_wires)
+                return qml.sample(wires=wires_m)
+
+        .. code-block:: pycon
+
+            >>> print(f"The ket representation of {m} + {k} mod {mod} is {adder_modulo(m, k, mod,wires_m, work_wires)}")
+            The ket representation of 8 + 5 mod 15 is [1 1 0 1]
+
+        We can see that the result [1 1 0 1] corresponds to 13, which comes from :math:`8+5=13 \longrightarrow 13 \, \text{mod} \, 15 = 13`.
+    """
+
+    grad_method = None
+
+    def __init__(
+        self, k, wires, mod=None, work_wires=None, id=None
+    ):  # pylint: disable=too-many-arguments
+        if mod is None:
+            mod = 2 ** (len(wires))
+        if work_wires is not None:
+            if any(wire in work_wires for wire in wires):
+                raise ValueError("None of the wires in work_wires should be included in wires.")
+        else:
+            work_wires = [wires[-1] + 1, wires[-1] + 2]
+        self.hyperparameters["k"] = k
+        self.hyperparameters["mod"] = mod
+        self.hyperparameters["work_wires"] = qml.wires.Wires(work_wires)
+        super().__init__(wires=wires, id=id)
```

Comment:
I have seen in the presentation that the auxiliary wires are not drawn when you do `qml.draw`. I think you should pass here the total wires

Reply:
Do you mean to do `wires`=`all_wires`, with `all_wires`=`wires`+`work_wires`? If we do that we should change the decomposition function, since `wires` will include all the wires